my-vault/internal/handlers/handlers.go

```diff
@@ -0,0 +1,199 @@
+	if err!=nil {
+		return domain.Secret{},err
+	
+	}
+
+	return domain.Secret{
+		Id:    uuid,
+		Name:  s.Name,
+		KVMap: s.KVMap,
+	},nil
+}
+
+func BuildListSecretsResponse(s []domain.Secret) ListSecretsResponse {
+	return ListSecretsResponse{
+		Message: "Secrets fetched successfully",
+		Secrets: s,
+	}
+}
+
+func (h *SecretHandler) HandlePostSecret(w http.ResponseWriter, r *http.Request) {
+
+	//var secret Secret
+	var secretRequestObject CreateSecretRequest
+
+	err := json.NewDecoder(r.Body).Decode(&secretRequestObject)
+	defer r.Body.Close()
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		log.Printf("ERROR: Unexpected behaviour: %v", err)
+		return
+	}
+	// build new secretObject
+	temp,err := NewCreateSecret(secretRequestObject)
+	if err!=nil {
+		secretErr := errors.New("unexpected internal error")
+		http.Error(w, secretErr.Error(), http.StatusInternalServerError)
+		log.Printf("ERROR: Unexpted internal error: %v", err)
+		return
+	}
+	//err = db.PutNewSecret(temp)
+	err = h.DB.PutNewSecret(temp)
+
+	if err != nil {
+		err := errors.New("unexpected internal error")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		log.Printf("ERROR: Unexpted internal error: %v", err)
+		return
+	}
+
+	res := CreateSecretResponse{
+		Id:      temp.Id,
+		Name:    temp.Name,
+		Message: fmt.Sprintf("Successfully created secret with secretId: %s and name: %s", temp.Id, temp.Name),
+	}
+	
+	log.Printf("INFO: Secret saved----------------------------------------- %s", temp.Id)
+	w.Header().Set("Content-Type", "application/json")
+	encoder := json.NewEncoder(w)
+	err = encoder.Encode(res)
+	if err != nil {
+
+		userErr := errors.New("unexpected internal error")
+		http.Error(w, userErr.Error(), http.StatusInternalServerError)
+		log.Printf("ERROR: Unexpted internal error: %v", err)
+		return
+	}
+}
+
+func (h *SecretHandler) HandleGetSecretById(w http.ResponseWriter, r *http.Request) {
+	//curl  -X POST -H "Content-Type: application/json" http://localhost:9000/secrets/234
+	secretID := r.PathValue("secret_id")
+	secretEntry, err := h.DB.GetScretsById(secretID)
+	log.Printf("secret from Db %v", secretEntry)
+	if err != nil {
+		if errors.Is(err, storage.ErrNotFound) {
+			http.Error(w, fmt.Sprintf("Secret with ID %s not found", secretID), http.StatusNotFound)
+			return
+		}
+		log.Printf("ERROR: Unexpected internal error: %v", err)
+		http.Error(w, "Unexpected internal error", http.StatusInternalServerError)	
+	return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	getSecretResponse := NewSecretResponseObject(secretEntry)
+	json.NewEncoder(w).Encode(getSecretResponse)
+	fmt.Printf("Retrieved secret----------------------------------------- %s", getSecretResponse)
+}
+
+func (h *SecretHandler) HandleDeleteSecretById(w http.ResponseWriter, r *http.Request) {
+	//curl  -X POST -H "Content-Type: application/json" http://localhost:9000/secrets/234
+	secretId := r.PathValue("secret_id")
```

Comment:
May want to check if this value is empty before continuing
```suggestion
	secretId := r.PathValue("secret_id")
	if len(secretId) == 0 {
		http.Error(w, "empty secret ID", http.StatusBadRequest)
	}
```